### DISCUSSION TITLE
pre request script  headers

### DISCUSSION THREAD

**Discussion by akhil-nuox** [hoppscotch/hoppscotch#3741] [Ideas]
How can we achieve something like this ?
`pm.request.headers.add({ key: 'Authorization', value: 'Bearer YOUR_ACCESS_TOKEN' });`

**Comment by jamesgeorge007** [comment]
Hi @akhil-nuox, we've introduced a `hopp` namespace housing new [scripting methods](https://docs.hoppscotch.io/documentation/features/scripts#hopp-%5Bexperimental%5D) as proposed in https://github.com/hoppscotch/hoppscotch/discussions/5221 with the ability to manage [requests](https://docs.hoppscotch.io/documentation/features/scripts#hopp-request-setheader-name%3A-string%2C-value%3A-string) from the scripting context.

It's not recommended to migrate or rework your existing scripts for the new scripting APIs just yet. Breaking changes are expected as the system evolves.

The rollout is intentionally gradual to gather feedback and refine based on real-world use. All future changes will be scoped to the experimental scripting sandbox (enabled by default; can be toggled in `Settings → Experiments`). Please feel free to take it for a spin and let us know your feedback via the RFC above.
